integration_tests/testsuite/deploy_app.py

```diff
@@ -102,37 +156,56 @@ def deploy_app(base_image, builder_image, appdir, yaml):
         if builder_image:
             _set_builder_image(builder_image)
 
-        deployed_version = test_util.generate_version()
-
-        # TODO: once sdk driver is published, use it here
-        deploy_command = ['gcloud', 'app', 'deploy', '--no-promote',
-                          '--version', deployed_version, '-q']
-        if yaml:
-            logging.info(yaml)
-            deploy_command.append(yaml)
-
-        subprocess.check_output(deploy_command)
+        if environment == constants.GAE_FLEX:
+            return deploy_app_gae(yaml)
+        elif environment == constants.GKE:
+            return deploy_app_gke(yaml)
 
-        return deployed_version
     except subprocess.CalledProcessError as cpe:
         logging.error('Error encountered when deploying application! %s',
                       cpe.output)
-        sys.exit(1)
-
+        raise cpe
+    except Exception as e:
+        logging.error('Error encountered when deploying application! %s', e)
+        raise e
```

Comment:
Can just do `raise` here (and anywhere else that you're re-raising the exception you just caught).

Reply:
fixed